### PR TITLE
Feature - download langs from google play store - android

### LINKED ISSUE #945: Android - Implement download language from google play store
We want to be able to redirect the user to the google play store in order to install specific TTS engines. Below are some design rules: 

- the feature is JUST for Android. 
- We have to create a json data file with the list of the TTS engines to be supported. 
- For every Engine, the list should contain:

  - engine name 
  - Languages supported 
  - Play store URL 
  - Vendor 

- We have to use a plugin to open the play store link https://github.com/xmartlabs/cordova-plugin-market
- A warning dialog must pop up before opening the play store link.
- In order to test the feature, below are some play store engines: 

   - https://play.google.com/store/apps/details?id=alfanum.co.rs.alfanumtts
   - https://play.google.com/store/apps/details?id=alfanum.co.rs.alfanumtts.cro
   - https://play.google.com/store/apps/details?id=alfanum.co.rs.alfanumtts.mne
   - https://play.google.com/store/apps/details?id=com.github.olga_yakovleva.rhvoice.android
   - https://play.google.com/store/apps/details?id=com.svox.classic

Below is the look and feel of the UI:

Captura de pantalla 2021-08-06 125837

Captura de pantalla 2021-08-06 131313

### PR DESCRIPTION
**Pull request by tomivm**
the messages aren't on the translation cboard.json file yet

### CODE REVIEW

**Comment by tomivm** [comment]
[here it is a Drive folder](https://drive.google.com/drive/u/0/folders/1Y5Y8no0kEAGXqcZlg-0GXab2-uyt9yIG) that contain: 
-APK to debug
-Screenshot of the future
-Use Cases documentation (need translation).

PLEASE consider rewrite dialogs messages.

File: src/components/Settings/Language/Language.container.js [hunk 1]
```diff
@@ -91,6 +109,249 @@ export class LanguageContainer extends Component {
...
+    }
+    const downloadingLangState = {
+      isdownloading: true,
+      isDiferentTts: true,
+      engineName: ttsName,
+      marketId: marketId,
+      selectedLang: lang
+    };
+    setDownloadingLang(downloadingLangState);
+    await this.handleSetTtsEngine(ttsName); //after tts change it fires a remounting
```
Comment (RodriSanchez1) [inline src/components/Settings/Language/Language.container.js]: this does not happen if I select a language from the downloadable list and have no voice available in the corresponding tts.

Comment (tomivm) [reply src/components/Settings/Language/Language.container.js]: good eye. thanks :)

File: src/components/Settings/Language/DownloadDialog.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  open: PropTypes.bool.isRequired,
+  onClose: PropTypes.func.isRequired,
+  onDialogAcepted: PropTypes.func.isRequired,
+  downloadingLangData: PropTypes.object
+};
+
+function DownloadDialog(props) {
+  const { open, onClose, onDialogAcepted, downloadingLangData, intl } = props;
+
+  const handleDialogAcepted = () => {
```
Comment (martinbedouret) [inline src/components/Settings/Language/DownloadDialog.js]: name should be `handleDialogAccepted`

File: src/components/Settings/Language/Language.component.js [hunk 3]
```diff
@@ -173,6 +188,29 @@ class Language extends React.Component {
...
+      return (
+        <Button
+          variant="outlined"
+          color="primary"
+          label="download"
+          onClick={event =>
+            this.props.onDownloadableLangClick(event, isDownloadable[0].id)
+          }
+        >
+          Download
```
Comment (martinbedouret) [inline src/components/Settings/Language/Language.component.js]: use translation texts

File: src/components/Settings/Language/Language.component.js [hunk 4]
```diff
@@ -220,123 +267,228 @@ class Language extends React.Component {
...
+              {!availableTts && (
+                <Chip label="unninstaled" size="small" disabled={false} />
+              )}
+            </div>
+            {!availableTts && (
+              <div className="Language__RightContent">
+                <Button
+                  variant="outlined"
+                  color="primary"
+                  label="download"
```
Comment (martinbedouret) [inline src/components/Settings/Language/Language.component.js]: use translation text

File: src/components/Settings/Language/Language.messages.js [hunk 5]
```diff
@@ -181,5 +185,59 @@ export default defineMessages({
...
+    defaultMessage: 'Download'
+  },
+  downloadDialogTitle: {
+    id: 'cboard.components.Settings.Language.downloadDialogTitle',
+    defaultMessage: 'Download Language'
+  },
+  downloadDialogSubtitle: {
+    id: 'cboard.components.Settings.Language.downloadDialogSubtitle',
+    defaultMessage:
+      'you will be redirect to the Google playstore in order to download and install the language you have chosen. Do you want to proced?'
```
Comment (martinbedouret) [inline src/components/Settings/Language/Language.messages.js]: You will be redirected to the Google play store in order to download and install the voices for this language. Do you want to proceed?'

File: src/components/Settings/Language/Language.messages.js [hunk 6]
```diff
@@ -181,5 +185,59 @@ export default defineMessages({
...
+  },
+  cancel: {
+    id: 'cboard.components.Settings.Language.cancel',
+    defaultMessage: 'Cancel'
+  },
+  ttsErrorAlert: {
+    id: 'cboard.components.Settings.Language.ttsErrorAlert',
+    defaultMessage: "Cboard didn't detect the new app. please Checkit"
```
Comment (martinbedouret) [inline src/components/Settings/Language/Language.messages.js]: We didn't detect the new language is installed. Please verify.

File: src/components/Settings/Language/Language.messages.js [hunk 7]
```diff
@@ -181,5 +185,59 @@ export default defineMessages({
...
+  },
+  langErrorAlert: {
+    id: 'cboard.components.Settings.Language.langErrorAlert',
+    defaultMessage:
+      "Cboard didn't detect the new voice, please open the app in order to download it"
```
Comment (martinbedouret) [inline src/components/Settings/Language/Language.messages.js]: We didn't detect the new voice is installed. Please verify the downloaded application.

File: src/components/Settings/Language/Language.messages.js [hunk 8]
```diff
@@ -181,5 +185,59 @@ export default defineMessages({
...
+    id: 'cboard.components.Settings.Language.CancelInstalation',
+    defaultMessage: 'Cancel instalation'
+  },
+  openApp: {
+    id: 'cboard.components.Settings.Language.openApp',
+    defaultMessage: 'Open app'
+  },
+  unninstaledLangNotification: {
+    id: 'cboard.components.Settings.Language.unninstaledLangNotification',
+    defaultMessage: 'install language to can use it'
```
Comment (martinbedouret) [inline src/components/Settings/Language/Language.messages.js]: Install the new language

File: src/components/Settings/Language/Language.messages.js [hunk 9]
```diff
@@ -181,5 +185,59 @@ export default defineMessages({
...
+  },
+  instaledLangSuccesNotification: {
+    id: 'cboard.components.Settings.Language.instaledLangSuccesNotification',
+    defaultMessage: 'Language sucefully instaled and seted'
```
Comment (martinbedouret) [inline src/components/Settings/Language/Language.messages.js]: Language installed successfully

**Review by martinbedouret — CHANGES_REQUESTED** [review]
Terrific job @tomivm ! My comments are just for the translation strings

**Review by martinbedouret — APPROVED** [review]

**Pull request merged**